bumpversion/cli.py

```diff
@@ -367,40 +389,56 @@ def main(original_args=None):
             replace=known_args.replace,
             part_configs=part_configs,
         )
-    except sre_constants.error as e:
+    except sre_constants.error:
         # TODO: use re.error here mayhaps, also: should we log?
         sys.exit(1)
+    return vc
+
+
+def _parse_current_version(current_version, vc):
+    if not current_version:
+        return None
+
+    return vc.parse(current_version)
+
 
-    current_version = (
-        vc.parse(known_args.current_version) if known_args.current_version else None
+def _assemble_context(vcs_info):
+    context = dict(
```

Comment:
I think this also works:

```suggestion
    context = dict(itertools.chain(time_context.items(), prefixed_environ().items(), vcs_info.items()))
```

Reply:
Good one, will include this.

Reply:
Python 3 has [ChainMap](https://docs.python.org/3/library/collections.html#collections.ChainMap) as well, but probably better to keep the codebase the same until we drop Python 2.